models/src/test/scala/coop/rchain/models/SortedHashSetSpec.scala

```diff
@@ -0,0 +1,65 @@
+package coop.rchain.models
+
+import java.io.ByteArrayOutputStream
+import java.util
+
+import com.google.protobuf.CodedInputStream
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.models.Expr.ExprInstance.GInt
+import coop.rchain.models.Var.VarInstance.BoundVar
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.models.rholang.sort.ordering._
+import org.scalatest.{Assertion, FlatSpec, Matchers}
+
+import scala.collection.immutable.BitSet
+
+class SortedHashSetSpec extends FlatSpec with Matchers {
+
+  // toByteArray method is using the same method calls as real protobuf's serialization
+  private[this] def serializeESet(parTreeSet: SortedHashSet[Par]): Array[Byte] =
+    ESet(ps = parTreeSet).toByteArray
```

Comment:
When serializing scalapb calls `writeTo(output: CodedOutputStream)` method on companion object but this `toByteArray` does the same and returns byte array of the output stream.